core/src/main/scala/filodb.core/memstore/ratelimit/CardinalityManager.scala

```diff
@@ -31,6 +32,22 @@ class CardinalityManager(datasetRef: DatasetRef,
   // physical resources for duplicate calculation
   private var isCardTriggered: Boolean = false
 
+
+  /**
+   * `dataset-configs` is an string array where each string is a file path for a dataset config. This function reads
+   * those file paths and parses it to Config object
+   * @param filoConfig
+   * @return
+   */
+  private def getDataSetConfigs(filoConfig: Config): Seq[Config] = {
+    val datasetConfPaths = filoConfig.as[Seq[String]]("dataset-configs")
+    if (datasetConfPaths.nonEmpty) {
+      datasetConfPaths.map { d => ConfigFactory.parseFile(new java.io.File(d)) }
+    } else {
+      Seq()
+    }
```

Comment:
`datasetConfPaths.map { d => ConfigFactory.parseFile(new java.io.File(d)) }` should be all that's needed here, I think

Reply:
Ack, updated :)